LibraBFT/Impl/Consensus/SafetyRules/SafetyRules.agda

```diff
@@ -247,5 +247,33 @@ signProposalM blockData = do
 
 ------------------------------------------------------------------------------
 
-postulate
-  signTimeoutM : Timeout → LBFT (Either ErrLog Signature)
+signTimeoutM : Timeout → LBFT (Either ErrLog Signature)
+signTimeoutM timeout = do
+ vs ← use (lSafetyRules ∙ srValidatorSigner)
+ maybeS vs (bail fakeErr) {-"srValidatorSigner", "Nothing"-} $ λ validatorSigner → do
+   safetyData ← use (lPersistentSafetyStorage ∙ pssSafetyData)
+   verifyEpochM (timeout ^∙ toEpoch) safetyData ∙^∙ withErrCtx (here' []) ∙?∙ λ _ -> do
+     ifM‖ timeout ^∙ toRound ≤? safetyData ^∙ sdPreferredRound ≔
+          bail fakeErr
+          --(ErrIncorrectPreferredRound (here []) (timeout^.toRound) (safetyData^.sdPreferredRound))
+        ‖ timeout ^∙ toRound <? safetyData ^∙ sdLastVotedRound ≔
+          bail fakeErr
+          --(ErrIncorrectLastVotedRound (here []) (timeout^.toRound) (safetyData^.sdLastVotedRound))
+        ‖ timeout ^∙ toRound >? safetyData ^∙ sdLastVotedRound ≔
+          verifyAndUpdateLastVoteRoundM (timeout ^∙ toRound) safetyData
+            ∙^∙ withErrCtx (here' [])
+            ∙?∙ (λ safetyData1 → do
+            lSafetyData ∙= safetyData1
```

Comment:
If I've understood correctly, we define the `lSafetyData` Lens only because in the Haskell code, the (equivalent) `pssSafetyData` Lens is derived using the `RWPersistentSafetyStorage` capability, which we're not modeling in the Agda implementation.  If I've got that right, let's spell `lSafetyData` as `pssSafetyData<something>` and put a comment noting this relationship next to its definition.  `<something>` could be just `'` or `equiv` or <something else> :smile:.

Reply:
My preference for naming is `pssSafetyData-rm`, indicating that it is akin to the `pssSafetyData` of the Haskell code with the type for the state instantiated with `RoundManager`.

Reply:
That works for me.